commands/init_test.go

```diff
@@ -1,32 +1,92 @@
 package commands
 
 import (
-	"github.com/github/hub/Godeps/_workspace/src/github.com/bmizerany/assert"
-	"github.com/github/hub/github"
+	"path/filepath"
+	"fmt"
 	"os"
-	"regexp"
 	"testing"
+
+	"github.com/github/hub/Godeps/_workspace/src/github.com/bmizerany/assert"
+	"github.com/github/hub/github"
 )
 
-func TestTransformInitArgs(t *testing.T) {
+func setup() {
```

Comment:
> Is there a way to have a setup block be defined only in the current file, for example with a func and a closure?

Unfortunately not.  Method name is unique for the same package, regardless which file it's in.

> Also, our current setup edits an environment variable but doesn't reset it at the end of the test.

That's no good. It'd be nice to reset the mutated global states if possible in tests. We've similar cases for fixture where we rest states: https://github.com/github/hub/blob/af9163b699b34ffa167ef99853ff2f7374d7b01a/git/git_test.go#L12-L13.